src/realm/table_view.hpp

```diff
@@ -407,8 +405,23 @@ class ConstTableView : public ObjList {
     size_t m_end = size_t(-1);
     size_t m_limit = size_t(-1);
 
+    class KeyValues : public KeyColumn {
+    public:
+        KeyValues()
+            : KeyColumn(Allocator::get_default())
+        {
+        }
+        KeyValues(const KeyValues&) = delete;
+        ~KeyValues()
+        {
+            destroy();
```

Comment:
`KeyValues` should behave just as std::vector<ObjKey>. That is, using memory from the heap and releasing memory when destroyed. Ideally I would just have used std::vector<ObjKey>, but it had some other problems down the line.

Reply:
You know, in the good old days we wanted to use columns everywhere instead of std::vector.

Reply:
Why do we want it instead of `std::vector`? It ties the lifetime of a database object to the lifetime of an in-memory object/function, which is what `DestroyGuard` is meant to solve. I'm confused about why such an object would appear as a member variable in `TableView`?

Reply:
Again - this does not represent a database object. It is used because we have a tradition to use Realm datatypes over std datatypes.

Reply:
OK, so I guess my feedback is that we shouldn't do that. 😛